harvard-lts/pdiiif
# Ltsviewer 225

**Adding healthcheck.*
* * *

**JIRA Ticket**: [LTSVIEWER-225](https://jira.huit.harvard.edu/browse/LTSVIEWER-225)

# What does this Pull Request do?
Adds an executable which can be used instead of curl to verify that the healthcheck endpoint is reachable.

# How should this be tested?

Build pdiiif off of the `LTSVIEWER-225` branch

You can go here in a browser and see the healthcheck with status 200:
http://localhost:23024/healthcheck/

Additionally, to test this now locally, you should be able to verify that from project root directory (while the pdiiif service is running):

From your host machine:
- Running `./pdiiif-api/healthcheck.cjs http://localhost:23024/healthcheck` => `HEALTHCHECK STATUS: 200`
- Running `./pdiiif-api/healthcheck.cjs http://localhost:23024/healthcheck-random-string` => `HEALTHCHECK STATUS: 404`

From the container (using `docker exec -it mps-pdiiif bash`):
- Running `./healthcheck.cjs http://mps-pdiiif:8080/healthcheck` => `HEALTHCHECK STATUS: 200`
- Running `./healthcheck.cjs http://mps-pdiiif:8080/healthcheck-random-string` => `HEALTHCHECK STATUS: 404`

While the service is not running, any of the host machine commands will return something like:

```
Error in Completing Health Check Request
Error: connect ECONNREFUSED 127.0.0.1:23024
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 23024
}
```

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@enriquediaz @f8f8ff
